File: android/patcher/src/main/kotlin/com/guthyerrz/autoproxy/patcher/util/Logger.kt
package com.guthyerrz.autoproxy.patcher.util

object Logger {

    private const val RESET = "\u001B[0m"
    private const val GREEN = "\u001B[32m"
    private const val YELLOW = "\u001B[33m"
    private const val RED = "\u001B[31m"
    private const val CYAN = "\u001B[36m"

    fun info(message: String) {
        println("${GREEN}[✓]${RESET} $message")
    }

    fun warn(message: String) {
        println("${YELLOW}[!]${RESET} $message")
    }

    fun error(message: String) {
        System.err.println("${RED}[✗]${RESET} $message")
    }

    fun step(message: String) {
        println("${CYAN}[→]${RESET} $message")
    }
}

File: ios/patcher/Sources/AutoProxyPatcher/Logger.swift
import Foundation

enum Logger {
    static var quiet = false

    private static let reset  = "\u{001B}[0m"
    private static let green  = "\u{001B}[32m"
    private static let yellow = "\u{001B}[33m"
    private static let red    = "\u{001B}[31m"
    private static let cyan   = "\u{001B}[36m"

    static func info(_ message: String) {
        guard !quiet else { return }
        print("\(green)[✓]\(reset) \(message)")
    }

    static func warn(_ message: String) {
        guard !quiet else { return }
        print("\(yellow)[!]\(reset) \(message)")
    }

    static func error(_ message: String) {
        let stderr = FileHandle.standardError
        stderr.write(Data("\(red)[✗]\(reset) \(message)\n".utf8))
    }

    static func step(_ message: String) {
        guard !quiet else { return }
        print("\(cyan)[→]\(reset) \(message)")
    }
}

extension FileHandle: @retroactive TextOutputStream {
    public func write(_ string: String) {
        write(Data(string.utf8))
    }
}
